src/physics/gyro_average.cxx

```diff
@@ -89,15 +89,15 @@ const Field2D gyroPade1(const Field2D &f, const Field2D &rho, int flags) {
 
 const Field3D gyroPade2(const Field3D &f, BoutReal rho, int flags) {
   Field3D result = gyroPade1(gyroPade1(f, rho, flags), rho, flags);
-  mesh->communicate(result);
+  result.getMesh()->communicate(result);
   result = 0.5*rho*rho*Delp2( result );
   result.applyBoundary("dirichlet");
   return result;
 }
 
 const Field3D gyroPade2(const Field3D &f, const Field2D &rho, int flags) {
   Field3D result = gyroPade1(gyroPade1(f, rho, flags), rho, flags);
-  mesh->communicate(result);
+  result.getMesh()->communicate(result);
```

Comment:
Just a thought looking at this syntax I wonder if it would be worth introducing a free function `communicate(Field3D &var) { var.getMesh()->communicate(var); return;}` -- probably not in this PR though.

Reply:
I like the idea. Could tidy up user code, reduce uses of global mesh pointer, and allow handling of fields with different meshes in the same call (i.e. if `f` and `g` have different `fieldmesh`es, `communicate(f, g)` could handle calling `communicate` from each mesh).